### DISCUSSION TITLE
Error using atomWithStorage with getOnInit in Next.js: "localStorage is not defined"

### DISCUSSION THREAD

**Discussion by beingtyson** [pmndrs/jotai#2691] [General] [answered]
I am trying to use atomWithStorage in Jotai with getOnInit set to true in a Next.js environment, but I encounter an error:

"ReferenceError: localStorage is not defined"

Due to this error, my client application resets to the initial value every time I refresh the page, even though I change the state.

**Comment by dai-shi** [comment, marked as answer]
Don't use `getOnInit` with SSR framework, such as Next.js. It's for pure client apps.

**Reply by beingtyson** [reply]
@dai-shi 
thx!
If the getOnInit option is not used, the state values disappear when the browser is refreshed. How can this be resolved?

**Reply by beingtyson** [reply]
thx i solved using useEffect

**Reply by johannbuscail** [reply]
Hello, how have you solved it pls ?
